### PR TITLE
Convert NativeAnimatedInterpolationTest to Kotlin

### PR DESCRIPTION
**Pull request by fabioh8010**
## Summary:

This PR converts NativeAnimatedInterpolationTest.java to Kotlin as requested in [this issue](https://github.com/facebook/react-native/issues/37708).

## Changelog:

[INTERNAL] [CHANGED] - Convert NativeAnimatedInterpolationTest to Kotlin

## Test Plan:

1. Run `./gradlew :packages:react-native:ReactAndroid:test`.
2. All tests should pass.

### CODE REVIEW

**Comment by fabioh8010** [comment]
@cortinico and @mdvacca could you please review? Thanks!

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+/**
+ * Tests method used by [InterpolationAnimatedNode] to interpolate value of the input nodes.
+ */
+@RunWith(RobolectricTestRunner::class)
+class NativeAnimatedInterpolationTest {
+  private fun simpleInterpolation(
+    value: Double, input: DoubleArray, output: DoubleArray
+  ): Double {
+    return InterpolationAnimatedNode.interpolate(
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt:22]:
```suggestion
  ): Double = InterpolationAnimatedNode.interpolate(
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+/**
+ * Tests method used by [InterpolationAnimatedNode] to interpolate value of the input nodes.
+ */
+@RunWith(RobolectricTestRunner::class)
+class NativeAnimatedInterpolationTest {
+  private fun simpleInterpolation(
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt:19]: Let's move this private method at the bottom of the class after all the tests?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY
+      )
+    ).isEqualTo(5.0)
+  }
+
+  @Test
+  fun testInterpolateColor() {
+    val input = doubleArrayOf(0.0, 1.0)
+    val output = intArrayOf(-0x1000000, -0x10000)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt:131]: Not a big fan of this conversion.
Can we use instead:

```
0xFF000000.toInt()
```

and make sure the original hex values are preserved? (valid for the whole test)

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this @fabioh8010 

I've left some comments that needs to be addressed before we can merge this. Thanks for doing this work.

**Review by mdvacca — APPROVED** [review]
lgtm

**Review by cortinico — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**